### DISCUSSION TITLE
Major guest behavior/ride age/opening rides problem in large old parks

### DISCUSSION THREAD

**Discussion by CSEverett** [OpenRCT2/OpenRCT2#15825] [Feature requests]
I'm playing old, mature parks that have been in existence for 20+ years of in-game time, and have around 7000 guests with not very many rides built in the last 5 years or so. Most of the major rides are a decade or more old. Although just about everything gets refurbished, which 'should' reset ride age, every few years. But now every time I open a new ride, hundreds (as many as 600+) of guests immediately head for the new ride. I've had 110 people heading for a Twist. Which inevitably overloads the ride and the footpath system, resulting in overcrowding and hundreds of guests leaving the park. And the other part of the problem is that using the "renew all rides" cheat either before, after or both opening the ride doesn't prevent it. And I'm using at least "double tracked" (two tiles with a fence between) footpaths in these cases, and I'm even having overcrowding on "four tracked" footpaths. Whatever the fix is, the problem means opening any new ride in these types of parks usually requires a carefully planned replacement of multiple rides around the park. Probably should either require a change in guest behavior, or improvements to what the "renew all rides" cheat actually does.

Right now this problem results in major difficulty in adding any new rides to large, old parks. I've heard others have the same problem. Hopefully this will be fixed in the 3.5 release?

Some screenshots to illustrate the problem:
![Screen Shot 10-26-21 at 04 23 PM 001](https://user-images.githubusercontent.com/89952921/139491784-683d4587-5db6-4301-b621-187c8589638c.PNG)

![Screen Shot 10-26-21 at 04 23 PM](https://user-images.githubusercontent.com/89952921/139491795-ad92988b-20fc-4dcc-998e-e9fb39f5f3ad.PNG)
![Screen Shot 10-26-21 at 04 22 PM](https://user-images.githubusercontent.com/89952921/139491819-a59d8459-7ae8-4dba-805f-2270d0c905b1.PNG)

**Comment by ocalhoun6** [comment]
Seems to me like an issue that should be looked into improving ... especially since the NSF will be resulting in even bigger parks. (And building such a huge park will probably make it end up being a pretty old park by the time you're done building.) So this issue will likely be experienced more often by more people in the NSF.

Maybe by placing a limit on how many guests can be heading for any particular ride at any one time? Like when a guest decides to head for a ride, they first check the guest stats, and if 200 other guests are already heading for that ride, the guest will cancel and not decide to head for that ride. (Though maybe instead of one set cap, the maximum number should depend on ride type, since different rides have different throughputs. A big coaster would probably be able to handle those 200 guests with room to spare, but a crooked house would be ludicrously overwhelmed by the crowd.)

**Comment by CSEverett** [comment]
Ooh, I didn't realize that with the NSF this will be an issue. A big one.  It definitely depends on ride type. But than,  in my experience the ability of a new ride to handle an influx of guests is different from ride to ride even of the same type,.  And I've had some rides than can have 150 or even nearly 200 people on them simultaneously, Perhaps (internally) compute a ride capacity figure and base the cap on that?
